src/js/tech/middleware.js

```diff
@@ -0,0 +1,63 @@
+import Tech from './tech.js';
+import toTitleCase from '../utils/to-title-case.js';
+
+const middlewares = {};
+
+export function use(type, middleware) {
+  (middlewares[type] = middlewares[type] || []).push(middleware);
```

Comment:
I also think that "Type" should be renamed to `mimeType`. I am also wondering if it would make more sense for middleware to come first in the arguments order.  I want to `use()` `hls` for `x` type(s). I also thing that type should be able to be an array.